src/main/java/org/opensearch/security/privileges/PitAccessEvaluator.java

```diff
@@ -0,0 +1,344 @@
+                                          User user, SecurityRoles securityRoles, final String action,
+                                          IndexNameExpressionResolver resolver,
+                                          PrivilegesEvaluatorResponse presponse) throws InterruptedException {
+        List<String> permittedPits = new ArrayList<>();
+        List<String> pitIds = getAllPitIds();
+        // allow delete pit operation if there are no pits in the cluster ( response should be empty )
+        if(pitIds.isEmpty()) {
+            deletePitRequest.clearAndSetPitIds(pitIds);
+            presponse.allowed = true;
+            presponse.markComplete();
+        }
+        Map<String, String[]> pitToIndicesMap = OpenSearchSecurityPlugin.GuiceHolder.getPitService().getIndicesForPits(pitIds);
+        for (String pitId : pitIds) {
+            String[] indices = pitToIndicesMap.get(pitId);
+            HashSet<String> indicesSet = new HashSet<>(Arrays.asList(indices));
+            Set<String> allPermittedIndices = getPermittedIndices(indicesSet, clusterService, user,
+                    securityRoles, action, resolver);
+            // user should have permissions for all indices associated with PIT, only then add PIT ID as permitted PIT
+            if(isDebugEnabled) {
+                log.debug("Evaluating PIT : " + pitId );
+            }
+            if (allPermittedIndices.size() == indicesSet.size()) {
+                if(isDebugEnabled) {
+                    log.debug(" Permitting PIT : " + pitId);
+                }
+                permittedPits.add(pitId);
+            }
+        }
+        // If there are any PITs for which the user has access to, then allow operation otherwise fail.
+        if(permittedPits.size() > 0) {
+            deletePitRequest.clearAndSetPitIds(permittedPits);
+            presponse.allowed = true;
+            presponse.markComplete();
+        }
+        return presponse;
+    }
+
+    /**
+     * Handle access for PIT segments API
+     */
+    private PrivilegesEvaluatorResponse handleGetAllPitSegmentsAccess(PitSegmentsRequest pitSegmentsRequest, ClusterService clusterService,
+                                                                 User user, SecurityRoles securityRoles, final String action,
+                                                                 IndexNameExpressionResolver resolver,
+                                                                 PrivilegesEvaluatorResponse presponse) throws InterruptedException {
+        List<String> permittedPits = new ArrayList<>();
+        List<String> pitIds = getAllPitIds();
+        // allow pit segments operation if there are no pits in the cluster ( response should be empty )
+        if(pitIds.isEmpty()) {
+            pitSegmentsRequest.clearAndSetPitIds(pitIds);
+            presponse.allowed = true;
```

Comment:
The authorization on a request should not be decided by whether there are PITs in a cluster. It should always be evaluated with permissions instead. If the user does not have right permissions, the request should be denied no matter whether there are PITs in a cluster or how many PITs there are.

Reply:
This is for the case where there are no PITs in cluster. We do evaluate the permissions.